.github/workflows/tests.yml

```diff
@@ -0,0 +1,45 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - dev
+      - main
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: linux
+            runs-on: ubuntu-22.04
+            arch: x86_64
+
+    name: "${{ matrix.os }}-${{ matrix.arch }}"
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Install Nim
+        uses: iffy/install-nim@0ed218de08059d18507144e6aee70e6bad0bc148 # v5.0.3
+        with:
+          version: "binary:2.0.0"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Avoid rate limiting.
+
+      - name: Install Nimble dependencies
+        run: nimble --accept install --depsOnly
+
+      - name: Build
+        run: nimble build
+
+      - name: Run tests
+        run: ./test || exit 0 # TODO: Stop allowing tests to fail.
```

Comment:
Do tests fail now? Even if they do I think we can still merge prs until we fix all tests